meinberlin/apps/projects/api.py

```diff
@@ -39,43 +43,50 @@ def get_queryset(self):
         )
         return projects
 
+    def list(self, request, *args, **kwargs):
+        key = get_projects_cache_key(suffix=self.request.GET.get("status"))
```

Comment:
One thing I didn't take into account when I tested this is that if we cache by status here, what happens in the following scenario:
1. running projects get cached (let's say for 2h) at 13:00
2. a project starts at 13:30
Will the project then not be listed until the cache expires at 15:00 ? If so we probably won't be able to do it like this as users should find the project once it started running. What if we cache the unfiltered qs instead? I guess we can't filter it after retrieving it, right? We could use a very short cache time but then the caching only helps with bursts of users, not in general

Reply:
Very good point. At first glance I can't find a signal for projects that started. If we had such a signal we could just hook it to the cache invalidation.

But I isn't there a email notification for projects that started? :thinking:

Reply:
there are no signals I think as that would require some background task which constantly checks if projects started/ended. So it's done by the rather complicated logic with different serializers for each status. However, you are right that there are emails if a project starts, so maybe there is something like that ?